kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala

```diff
@@ -108,4 +109,20 @@ object HighAvailabilityConf {
     .version("1.4.0")
     .stringConf
     .createOptional
+
+  val HA_ZK_ENGINE_POOL_SIZE: ConfigEntry[Int] =
+    buildConf("ha.engine.pool.size")
+      .doc("Maximum number of engines when using USER share level.")
+      .version("1.4.0")
+      .intConf
+      .checkValue(s => s > 0 && s < 50, "Invalid Engine Pool Size.")
+      .createWithDefault(1)
+
+  val HA_ZK_ENGINE_PROVIDE_POLICY: ConfigEntry[String] =
+    buildConf("ha.engine.provide.policy")
+      .doc("Multiple engines can be exposed under same space when using USER share level, " +
+        "choose the appropriate strategy according to different scenarios.")
+      .version("1.4.0")
+      .stringConf
+      .createWithDefault(ProvidePolicy.RANDOM.toString)
```

Comment:
```
    .stringConf
    .transform(_.toUpperCase(Locale.ROOT))
    .checkValues(ProvidePolicy.values.map(_.toString))
```

Reply:
👍

Reply:
I think you missed `.checkValues(ProvidePolicy.values.map(_.toString))`